### DISCUSSION TITLE
Installation problem in virtual vm.

### DISCUSSION THREAD

**Discussion by Mhsh** [vernemq/vernemq#1872] [Q&A]
Hi, 
I am trying to install vernemq in virtual vm ubuntu. 
Env: Oracle VM virtual box - ubuntu-20.04.2.0-desktop-amd64.iso. 
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal

Going through the installation steps for 'Building VerneMQ from sources' from https://vernemq.com/downloads/

- Installed Erlang OTP 
- git clone git://github.com/erlio/vernemq.git - success
- make rel - getting error.

Below is the error 

Generating "include/crypto_compat.hrl" ...
...supports cryto:hash/2
...writing "include/crypto_compat.hrl"
===> Analyzing applications...
===> Compiling vernemq_dev
===> Compiling _build/default/lib/vernemq_dev/src/on_topic_unsubscribed_hook.erl failed
_build/default/lib/vernemq_dev/src/on_topic_unsubscribed_hook.erl:3:10: can't find include file "vernemq_dev.hrl"

_build/default/lib/vernemq_dev/src/on_topic_unsubscribed_hook.erl:6:52: type subscriber_id() undefined
_build/default/lib/vernemq_dev/src/on_topic_unsubscribed_hook.erl:11:8: type topic() undefined

make: *** [Makefile:26: rel] Error 1

I am not sure what to do next. 

Note: I am naive to linux.

**Comment by ioolkos** [comment]
@Mhsh Verne does not yet compile on OTP 24. Is that what you tried?
